docs/architecture/adr-009-non-interactive-default-rules-for-reduced-padding.md

```diff
@@ -0,0 +1,177 @@
+
+![Current ni rules proof size](./assets/current-ni-rules-proof-size.png)
+
+NMT-Node size := 32 bytes + 2\*8 bytes = 48 bytes  
+MT-Node size := 32 bytes
+
+Proof size = subtree roots (rows) + subtree roots (last row) + blue nodes (parity shares) + 2 \* blue nodes (`DataRoot`)  
+Proof size = (log(n) + log(k) + log(n)) \* NMT-Node size  + 2\*log(k) \* MT-Node size  
+Proof size = 48 \* (2\*log(n) + log(k)) + 64 \*log(k)
+
+### Proposed Non-Interactive Default Rules
+
+Each row consists of sqrt(n)/log(n) subtree roots. Which makes in total sqrt(n) subtree roots. The rest is here the same as before.
+
+![Proposed ni rules proof size](./assets/proposed-ni-rules-proof-size.png)
+
+Proof size = subtree roots (all rows) + subtree roots (last row) + blue nodes (parity shares) + 2 \* blue nodes (`DataRoot`)  
+Proof size = (sqrt(n) + log(k) + log(n)) \* NMT-Node size  + 2\*log(k) \* MT-Node size  
+Proof size = 48 \* (sqrt(n) + log(k) + log(n)) + 64 \*log(k)
+
+## 5. What is the worst constructible block with the most amount of padding with old and new non-interactive default rules?
+
+For the current non-interactive default rules, when you have a square size of k you can alternate messages of size k/2 +1 and k/4 +1 to have the most amount of padding. This forces you to always use a new row.
+
+Padding = (k/2 -1) \* (k/2 -1) + (3k/4 -1) \* (k/2 -1)
+
+To have the most amount of padding for the proposed non-interactive default rules you use repeated messages of size 5 which will result in a padding of 3 in between.
+
+Padding = 3 \* (k-1) \* k/8
+
+![Worst Case Padding](./assets/worst-case-padding.png)
+
+## What is the quantified padding and proof size cost?
+
+Proof size increases from 2928 bytes to 10352 bytes in 2 GB blocks. In the current `MaxSquareSize` it's from 2096 to 3088 bytes. For bigger messages, the number of row roots will approach sqrt(n). Before that, we will get to k/4 roots which will make the message act the same before and after the proposed non-interactive default rules.
+
+![Proof Size Result](./assets/proof-size-result.png)
+
+The worst-case padding decreases from 1.3 GB to 0.8 GB in 2 GB Blocks. In the current `MaxSquareSize` it's from 5 MB to 3 MB. In general, the worst-case padding size approaches in current non-interactive default rules 66% and the proposed non-interactive default rules 37.5%. That is a reduction of almost 50%.
+
+![Padding Size Result](./assets/padding-size-result.png)
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+
+The padding decreases a lot.
+
+### Negative
+
+The proof size increases a bit.
```

Comment:
[very optional]
I could see it as being orthogonal, but we might want to go into more of the indirect consequences, such as zk prover costs or possibly decreasing bandwith for trust minimized light clients by decreasing the average DAH size (assuming the same amount of data posted).